metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala

```diff
@@ -8,8 +8,13 @@ import org.eclipse.lsp4j.TextEdit
 
 object IndentOnPaste extends RangeFormatter {
 
+  private val noEndClause = raw"((<!\bend\b\s*?)\b(if|while|for|match|try))"
+  private val ifThenClause = raw"(\bif\s+(?!.*?\bthen\b.*?$$)[^\s]*?)"
+  private val keywordClause = raw"\b(then|else|do|catch|finally|yield|case)"
+  private val openBraceClause = raw"""(^.*\{[^}"']*$$)"""
+  private val extensionClause = raw"""extension\s*((\(|\[).*(\)|\]))+"""
```

Comment:
This allows to have an increase pattern also in the following case:
```scala
extension [T] \\ here it will increase the Intendation
  (using b: B)(s: String) 
```

will this be a valid syntax?

And I'm not sure that other regexes will handle the indentation after the second line, but as @tgodzik said, it doesn't handle multiple lines.

Reply:
Och, I think this should not increase the indentation, that's msot likely not a valid syntax.

Reply:
Actually, the syntax works, but it was not the intention :O

Reply:
Scratch that! Funnily enough, we should indent after anything because:
```scala
extension [T] \\ here it will increase the Intendation
  (using b: B)(s: String) 
```

is valid while:
```scala
extension [T] \\ here it will increase the Intendation
(using b: B)(s: String) 
```
is not.